lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1267,14 +1274,16 @@ def _create_deployment_parameters(
         )
 
         # change deployment for each feature.
-        for f in features_settings.values():
-            feature_type = next(
-                x for x in self.supported_features() if x.name() == f.type
-            )
+        for feature_type, setting in [
+            (t, s)
+            for t in self.supported_features()
+            for s in features_settings.values()
+            if t.name() == s.type
+        ]:
```

Comment:
Is this an acceptable solution? It will guarantee the order of on_before_deployment will match the order of the list returned by AzurePlatform.supported_features()
Do you foresee this change causing any problems? Is it a bad idea to rely on a guaranteed order of execution?

This felt like a better solution to me than trying to catch specific deployment errors in hooks.py

Reply:
Yes, please also leave comments here and `supported_features` to specify the order is respected, and write comments explicitly explain the hibernation depends on Availability in the feature list.